### ISSUE TITLE
Build error on MIPS 32bit

### ISSUE THREAD

**Issue by faulesocke** [async-rs/async-std#664]
```
error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
 --> [...]/async-std-1.4.0/src/task/task_id.rs:2:25
  |
2 | use std::sync::atomic::{AtomicU64, Ordering};
  |                         ^^^^^^^^^
  |                         |
  |                         no `AtomicU64` in `sync::atomic`
  |                         help: a similar name exists in the module: `AtomicU8
```

Unfortunately, according to the docs, 64bit atomic types are not available on 32bit mips architectures.

I've looked into the source location and two solutions to this come to my mind:

1. use AtomicU32. This might or might not be okay on 32bit architectures.
2. Rewrite this (for mips only) with blocking logic (i.e. keep the counter inside a mutex)

If someone tells me which one is the preferred solution, I can write a PR.

**Comment by yoshuawuyts** [comment]
Related: https://github.com/async-rs/async-std/pull/286

**Comment by faulesocke** [comment]
Yep, this looks like it will solve my problem. Any estimations on when it might be ready?

**Comment by yoshuawuyts** [comment]
@faulesocke this will mostly depend on availability of our team; we'll be sitting down to plan next steps for async-std next week, I'll be sure to bring this up.

**Comment by faulesocke** [comment]
It looks like #286 was closed without being merged. Are there other plans to solve this?
